Detectors/MUON/MCH/Tracking/src/TrackAtVertexSpec.cxx

```diff
@@ -0,0 +1,389 @@
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file TrackAtVertexSpec.cxx
+/// \brief Implementation of a data processor to extrapolate the tracks to the vertex
+///
+/// \author Philippe Pillot, Subatech
+
+#include "TrackAtVertexSpec.h"
+
+#include <chrono>
+#include <stdexcept>
+#include <list>
+
+#include <TMath.h>
+#include <TGeoManager.h>
+#include <TGeoGlobalMagField.h>
+
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Output.h"
+#include "Framework/Task.h"
+
+#include "DetectorsBase/GeometryManager.h"
+#include "MathUtils/Cartesian3D.h"
+#include "Field/MagneticField.h"
+#include "MCHBase/ClusterBlock.h"
+#include "MCHBase/TrackBlock.h"
+#include "TrackParam.h"
+#include "TrackExtrap.h"
+
+namespace o2
+{
+namespace mch
+{
+
+using namespace std;
+using namespace o2::framework;
+
+class TrackAtVertexTask
+{
+ public:
+  //_________________________________________________________________________________________________
+  void init(framework::InitContext& ic)
+  {
+    /// Prepare the track extrapolation tools
+
+    LOG(INFO) << "initializing track extrapolation to vertex";
+
+    if (!gGeoManager) {
+      o2::base::GeometryManager::loadGeometry();
+      if (!gGeoManager) {
+        throw runtime_error("cannot load the geometry");
+      }
+    }
+
+    if (!TGeoGlobalMagField::Instance()->GetField()) {
+      auto l3Current = ic.options().get<float>("l3Current");
+      auto dipoleCurrent = ic.options().get<float>("dipoleCurrent");
+      auto field = o2::field::MagneticField::createFieldMap(l3Current, dipoleCurrent, o2::field::MagneticField::kConvLHC, false,
+                                                            3500., "A-A", "$(O2_ROOT)/share/Common/maps/mfchebKGI_sym.root");
+      TGeoGlobalMagField::Instance()->SetField(field);
+      TGeoGlobalMagField::Instance()->Lock();
+      TrackExtrap::setField();
+    }
+
+    auto stop = [this]() {
+      LOG(INFO) << "track propagation to vertex duration = " << mElapsedTime.count() << " s";
+    };
+    ic.services().get<CallbackService>().set(CallbackService::Id::Stop, stop);
+  }
+
+  //_________________________________________________________________________________________________
+  void run(framework::ProcessingContext& pc)
+  {
+    /// read the tracks with attached clusters of the current event,
+    /// propagate them to the corresponding vertex and send the new version
+
+    // get the vertex
+    int eventVtx(-1);
+    Point3D<double> vertex(0., 0., 0.);
+    try {
```

Comment:
what's the use of the try...catch block if you don't actually do anything useful in the catch ?

Reply:
This is how I understand exceptions. I want to stop that function here if an exception is catch and throw this exception so that the top function can also catch it.

Reply:
I did some tests and indeed this try/catch in not needed here for the code to run as intended. So I removed it.